overlord/snapstate/conflict.go

```diff
@@ -98,6 +98,42 @@ func affectedSnaps(t *state.Task) ([]string, error) {
 	return nil, nil
 }
 
+func snapSetupFromChange(chg *state.Change) (*SnapSetup, error) {
+	for _, t := range chg.Tasks() {
+		// Check a known task of each change that we know keep snap info.
+		if t.Kind() != "prerequisites" {
+			continue
+		}
+		return TaskSnapSetup(t)
+	}
+	return nil, nil
+}
+
+// verifySnapdIsNotRefreshing verifies that no change is currently running
+// for a snapd refresh/revert. While snapd is updating we do not allow other
+// snap changes to be executing, to avoid any incompatibilities in the task system,
+// between snapd versions.
+func changeIsSnapdDowngrade(st *state.State, chg *state.Change) bool {
+	snapsup, err := snapSetupFromChange(chg)
+	if err != nil || snapsup == nil || snapsup.SnapName() != "snapd" {
+		return false
+	}
+
+	var snapst SnapState
+	err = Get(st, snapsup.InstanceName(), &snapst)
+	if err != nil {
+		return false
+	}
+
+	// If snapd was not installed, or if the revision we are installing
+	// is local do not consider this a downgrade. (If the revision currently
+	// install is local then the check below will return false anyway).
+	if !snapst.IsInstalled() || snapsup.Revision().Local() {
+		return false
+	}
+	return snapsup.Revision().N < snapst.Current.N
```

Comment:
sadly revisions are not enough to be sure that this is a downgrade because a newer revision can exist that holds an older version :/